cloudpickle/cloudpickle.py

```diff
@@ -141,22 +142,38 @@ def _whichmodule(obj, name):
     return None
 
 
-def _is_global(obj, name=None):
+if sys.version_info[:2] < (3, 7):
+    def _get_module_attr(obj):
```

Comment:
Honestly, I don't like adding such special cases for the sake of semi-bugs. The fact that the  `__module__` attribute of `TypeVars` is wrong in `Python < 3.7` is a bug of the typing module. I'm tempted to not fix this for  `Python < 3.7` e.g pickle module-importable `TypeVars` as dynamic. @ogrisel WDYT?

Reply:
I am not sure what you suggest.

Also I am not familiar with what is wrong with the value of `obj.__module__`  for instances of `TypeVar` in Python < 3.7.

On my Python 3.6 I get:

```python
>>> import typing
>>> typing.AnyStr.__module__
'typing'
>>> type(typing.AnyStr)
typing.TypeVar
```

This looks fine and behaves as in Python 3.8.

In any case, if we implement a workaround for a Python bug in cloudpickle we should always insert a comment that explains what is the bug (with a link to the CPython issue tracker) and how the workaround works (e.g. here treat the typevar instance as an instance of a dynamic typevar instead of an instance of a typevar that can be lookedup by name.

Reply:
I mean that if you create  `mod.py` and add `t  = typing.TypeVar('t')` in it,
and then, in an interactive session:
```python
from mod import t
t.__module__
```
outputs  `typing`, which is wrong. Hence the workaround on which I emitted some reservations.

Reply:
Ok thanks for the clarification. My take is then:

```
$ python -c "import this" | grep purity
Although practicality beats purity.
```

So +1 for keeping the workaround but we should better document the issue in an inline comment. I tried to google a bit to find a related ticket in in the CPython issue tracker but I could not find any...

Reply:
The problem is that this problem only affect `Python 3.5` and `Python 3.6` which are now in security mode, so I suspect CPython folks will discard this issue.

Reply:
Ok I won't bikeshed too much on this. I want to move forward with this series of PRs, so let's keep it as is, and I may refactor this `__module__` handling at some point in the future.